### PR TITLE
[FBZ-10448] - Migrate New Relic Infra Recipe to v7

### PR DESCRIPTION
**Pull request by kvbautista**
**Description of your patch**
Adds support for New Relic Infra by migrating recipe from v6 to v7

**Recommended Release Notes**
Added support for New Relic Infra

**Estimated risk**
Medium

**Components involved**
`cookbooks/ey-newrelic_infra`
`custom-cookbooks/newrelic_infra`

**Description of testing done**

* Create a new environment with v7 as stack
* Upload recipe for `cookbooks/ey-newrelic_infra` and `custom-cookbooks/newrelic_infra`
* Verify through the chef logs that ey-newrelic_infra recipe has been executed
* Verify if New Relic Package has been successfully installed

**QA Instructions**

* Create a new environment with v7 as stack
* Upload recipe for `cookbooks/ey-newrelic_infra` and `custom-cookbooks/newrelic_infra`
* Verify through the chef logs that ey-newrelic_infra recipe has been executed
* Verify if New Relic Package has been successfully installed

### CODE REVIEW

File: cookbooks/ey-newrelic_infra/recipes/default.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  group "root"
+  mode "0600"
+  backup false
+  variables({
+    license_key: license_key,
+    display_name: display_name,
+  })
+end
+
+execute "curl -s https://download.newrelic.com/infrastructure_agent/gpg/newrelic-infra.gpg | sudo apt-key add -" do
```
Comment (mushyy) [inline cookbooks/ey-newrelic_infra/recipes/default.rb]: Keys should be added under apt_repository process

File: cookbooks/ey-newrelic_infra/files/default/newrelic-infra.list [hunk 2]
```diff
@@ -0,0 +1 @@
+deb [arch=amd64] https://download.newrelic.com/infrastructure_agent/linux/apt focal main
```
Comment (mushyy) [inline cookbooks/ey-newrelic_infra/files/default/newrelic-infra.list]: This file should be generated when adding also, the type should support arm64 as well as amd64

**Review by mushyy — CHANGES_REQUESTED** [review]

File: cookbooks/ey-newrelic_infra/recipes/default.rb (hunk 1 above)
Comment (kvbautista) [reply cookbooks/ey-newrelic_infra/recipes/default.rb]: Moved key under apt_repository process

```
apt_repository "newrelic-infra" do
  uri "https://download.newrelic.com/infrastructure_agent/linux/apt"
  distribution "focal"
  distribution "#{`lsb_release -cs`.strip}"
  components ["main"]
  key "https://download.newrelic.com/infrastructure_agent/gpg/newrelic-infra.gpg"
end.run_action(:add)
```

File: cookbooks/ey-newrelic_infra/files/default/newrelic-infra.list (hunk 2 above)
Comment (kvbautista) [reply cookbooks/ey-newrelic_infra/files/default/newrelic-infra.list]: Removed files folder as `newrelic-infra.list` should be generated upon execution of `apt_repository` process

**Comment by kvbautista** [comment]
Moved key under apt_repository process for `cookbooks/ey-newrelic_infra/recipes/default.rb`

```
apt_repository "newrelic-infra" do
  uri "https://download.newrelic.com/infrastructure_agent/linux/apt"
  distribution "focal"
  distribution "#{`lsb_release -cs`.strip}"
  components ["main"]
  key "https://download.newrelic.com/infrastructure_agent/gpg/newrelic-infra.gpg"
end.run_action(:add)
```

And removed files folder as newrelic-infra.list should be generated upon execution of `apt_repository` process.

**Review by mushyy — COMMENTED** [review]
Looks good to me passes linting process

**Comment by mushyy** [comment]
This will be merged for next release

**Review by mushyy — APPROVED** [review]

**Pull request merged**